File: 0949-58-2559-count-vowel-strings-in-ranges/0949-58-2559-count-vowel-strings-in-ranges.js
var vowelStrings = function(words, queries) {
    const isVowel = (ch) => 'aeiou'.includes(ch);

    const n = words.length;
    const pre = Array(n).fill(0);

    for (let i = 0; i < n; i++) {
        pre[i] = isVowel(words[i][0]) && isVowel(words[i][words[i].length - 1]) ? 1 : 0;
        if (i > 0) {
            pre[i] += pre[i - 1];
        }
    }

    return queries.map(([l, r]) => pre[r] - (l > 0 ? pre[l - 1] : 0));
};

File: 0949-58-2559-count-vowel-strings-in-ranges/0949-58-2559-count-vowel-strings-in-ranges.cpp
class Solution {
public:
    inline bool inVowel(char ch){
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'u' || ch == 'o';
    }
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        int n = words.size();
        vector<int>pre(n);
        for(int i = 0 ; i < n ; i++){
            pre[i] = inVowel(words[i][0]) & inVowel(words[i].back());
            if(i)pre[i]+=pre[i-1];
        }
        int q = queries.size();
        vector<int>ans(q);
        for(int i = 0 ; i < q;i++){
            int l = queries[i][0], r = queries[i][1];
            ans[i] = pre[r] - (l ? pre[l-1] : 0);
        }
        return ans;

    }
};